stdlib/parray.mc

```diff
@@ -0,0 +1,369 @@
+
+include "string.mc"
+
+-- This file contains an implementation of pure arrays, a way to use random access in
+-- arrays in a pure setting. The operations have low constant factor and a time complexity
+-- of O(log n).
+
+-- Pure Array
+type PA a
+con PAData  : all a. (Int,a,a,a,a,a,a,a,a,a,a) -> PA a
+con PANode  : all a. (Int,Int,PA a,PA a,PA a,PA a,PA a,PA a,PA a,PA a,PA a,PA a) -> PA a
+con PAEmpty : all a. () -> PA a
+con PANext  : all a. (PA a) -> PA a
+
+let empty = PAEmpty()
```

Comment:
This should be renamed to avoid name collisions

Reply:
Fixed in next version.